plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/AppEngineComponentPage.java

```diff
@@ -33,19 +32,23 @@
  * with instructions on how to install it. This page disables the 'Finish' button.
  */
 public class AppEngineComponentPage extends WizardPage {
+  private boolean forNativeProjectWizard;
 
-  protected AppEngineComponentPage() {
+  public AppEngineComponentPage(boolean forNativeProjectWizard) {
     super("appEngineComponentPage");
     setTitle("App Engine Component is missing");
     setDescription("The Cloud SDK App Engine Java component is not installed"); 
+    this.forNativeProjectWizard = forNativeProjectWizard;
   }
 
   @Override
   public void createControl(Composite parent) {
     AnalyticsPingManager.getInstance().sendPing(
         AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD,
         AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE,
-        AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_NATIVE, parent.getShell());
+        forNativeProjectWizard ? AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_NATIVE :
+          AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_MAVEN,
+          parent.getShell());
```

Comment:
Formatting is weird. `AnalyticsEvents.APP_ENGINE_NEW_PROJECT_WIZARD_TYPE_MAVEN,` should be indented (perhaps `:` should come before it?) while `parent.getShell()` should be de-indented.